src/main.c

```diff
@@ -79,50 +85,56 @@ int load_conf_file(char *file_path)
 static void run_tests()
 {
     print("Random seed used is %u", seed);
-    if(tests_to_run.size == 0) {
+    if (tests_to_run.size == 0) {
         print("No Specific tests specified. Running all tests (Single Pass).");
         print("-------------------------------------------------------------");
         int table_size = ARRAY_SIZE(kernel_thunk_table);
-        for(int k=0;k<table_size;k++){
+        for (int k=0; k<table_size; k++) {
             kernel_thunk_table[k]();
         }
     }
-    else{
+    else {
         print("Config File Was Loaded. Only running requested tests.");
         print("-----------------------------------------------------");
-        for(int k=0; k<tests_to_run.size; k++){
+        for (int k=0; k<tests_to_run.size; k++) {
             kernel_thunk_table[vector_get(&tests_to_run, k)]();
         }
     }
     print("------------------------ End of Tests -----------------------");
 }
 
-void main(void){
-
-    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+void main(void)
+{
+    vector_init(&tests_to_run);
+    if (!open_output_file("D:\\kernel_tests.log")) {
+        return;
+    }
+    // If load config fail, then do what need to be initialize.
+    if (load_conf_file("D:\\config.txt")) {
+        // NOTE: Pre-init is set to FALSE because of video initialization doesn't occur until below.
+        output_video = TRUE;
+    }
 
-    switch(pb_init()){
-        case 0: break;
-        default:
+    if (output_video) {
+        XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+        // If pb_init return non-zero, then something went wrong from nxdk's end.
```

Comment:
```suggestion
        // If pb_init returns non-zero, then something went wrong from nxdk's end.
```

Reply:
fixed, also added "the" and "call" in the sentence.